lib/galaxy/datatypes/tabular.py

```diff
@@ -53,26 +53,22 @@ def displayable( self, dataset ):
         except:
             return False
 
-    def get_chunk(self, trans, dataset, chunk):
-        ck_index = int(chunk)
-        f = open(dataset.file_name)
-        f.seek(ck_index * trans.app.config.display_chunk_size)
-        # If we aren't at the start of the file, seek to next newline.  Do this better eventually.
-        if f.tell() != 0:
+    def get_chunk(self, trans, dataset, offset=0, ck_size=0):
+        with open(dataset.file_name) as f:
+            f.seek(offset)
+            ck_data = f.read(ck_size or trans.app.config.display_chunk_size)
             cursor = f.read(1)
-            while cursor and cursor != '\n':
+            while cursor and cursor != '\n' and ck_data[-1] != '\n':
+                ck_data += cursor
                 cursor = f.read(1)
```

Comment:
This is not correct: when the chunk read at line 59 terminates with `'\n'`, the first character of the next line will be read at line 60 and then lost.
The correct version is:
```
    ck_data = f.read(chunk_size or trans.app.config.display_chunk_size)
    if ck_data and ck_data[-1] != '\n':
        cursor = f.read(1)
        while cursor and cursor != '\n':
            ck_data += cursor
            cursor = f.read(1)
```